canvas-discussion-bot/index.js

```diff
@@ -0,0 +1,137 @@
+const axios = require('axios');
+const dotenv = require('dotenv');
+const { faker } = require('@faker-js/faker');
+const fs = require('fs');
+const path = require('path');
+const puppeteer = require('puppeteer');
+
+async function retrieveCourses(apiToken, canvasURL) {
+  try {
+    const endPoint = "/api/v1/accounts/self/courses";
+    const url = canvasURL + endPoint;
+    
+    const response = await axios.get(url, {
+      headers: {
+        'Authorization': `Bearer ${apiToken}`
+      }
+    });
+    
+    return response.data;
+  } catch (error) {
+    throw new Error(`Error retrieving courses: ${error.message}`);
+  }
+}
+
+// Specify the path to the .env file for the local directory
+const envFilePath = path.resolve(__dirname, '.env');
+
+// Load the .env file
+const result = dotenv.config({ path: envFilePath });
+
+if (result.error) {
+  console.error('Error loading .env file:', result.error);
+} else {
+  console.log('Successfully loaded .env file');
+}
+
+async function postDiscussion(username, password, courseId, index) {
+  const browser = await puppeteer.launch({ headless: false });
+  const page = await browser.newPage();
+  console.log('postDiscussion Course Id:', courseId); 
+  try {
+    // Navigate to the Canvas login page
+    await page.goto(process.env.CANVAS_URL);
+
+    // Perform login
+    // Wait for the first textbox to become visible and type username
+    await page.waitForSelector('#pseudonym_session_unique_id_forgot');
+    await page.type('#pseudonym_session_unique_id_forgot', username);
+
+    // Wait for the second textbox to become visible and type password
+    await page.waitForSelector('#pseudonym_session_password');
+    await page.type('#pseudonym_session_password', password);
+
+    // Wait for the first button to become visible and click
+    await page.waitForSelector('.Button');
+    await page.click('.Button');
+    console.log('Hit the button')
+    // Wait for navigation to complete
+    await page.waitForNavigation();
+
+
+    // Navigate to course and post discussion
+    const courseURL = `${process.env.CANVAS_URL}/courses/${courseId}/discussion_topics`;
+    await page.goto(courseURL);
+    await page.waitForSelector('#add_discussion');
+    await page.click('#add_discussion'); 
+
+    await page.waitForSelector('#discussion-title');
+    await page.type('#discussion-title', faker.lorem.sentence());
+
+    await page.waitForSelector('#discussion-topic-message2_ifr');
+    const iframeElement = await page.$('#discussion-topic-message2_ifr');
+    const iframe = await iframeElement.contentFrame();
+    await iframe.waitForSelector('body');
+    await iframe.type('body', faker.lorem.sentences(8));
+
+    // Click the "Save" button
+    await page.waitForSelector('button.btn.btn-primary.submit_button');
+    await page.click('button.btn.btn-primary.submit_button');
+
+    // Add a 5-second delay
+    await new Promise(resolve => setTimeout(resolve, 5000));
```

Comment:
can probably just do 
```ts
    await Promise.all([
      page.waitForNavigation(),
      page.click('button.btn.btn-primary.submit_button')
    ]);
```
instead of delaying 5 seconds to wait for navigation and submission.

Reply:
Yeah, i don't need that all. That was a debugging delay, just so I could see how far it got.